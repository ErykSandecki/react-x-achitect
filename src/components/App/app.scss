@use 'sass:map';

.App {
  min-height: 100%;
  overflow: auto;

  @each $mode in $--cth-theme-list {
    &--#{$mode} {
      background-color: map.get($--cth-colors-map, 'neutral-4' + '--' + $mode);
      color: map.get($--cth-colors-map, 'neutral-1' + '--' + $mode);
    }
  }
}
