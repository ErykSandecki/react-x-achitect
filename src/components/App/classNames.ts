export const className = 'App';

export const classNames = {
  [className]: className,
};
