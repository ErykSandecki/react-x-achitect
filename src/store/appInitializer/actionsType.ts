export const REDUCER_KEY = 'appInitializer';

export const APP_INIT = `${REDUCER_KEY}/APP_INIT`;
export const INIT_LANGUAGE = `${REDUCER_KEY}/INIT_LANGUAGE`;
export const INIT_LANGUAGE_SUCCESS = `${REDUCER_KEY}/INIT_LANGUAGE_SUCCESS`;
export const SET_IS_APP_LOADED = `${REDUCER_KEY}/SET_IS_APP_LOADED`;
export const SET_THEME = `${REDUCER_KEY}/SET_THEME`;
