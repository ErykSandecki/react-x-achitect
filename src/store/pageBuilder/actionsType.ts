export const REDUCER_KEY = 'pageBuilder';

export const ADD_ELEMENT = `${REDUCER_KEY}/ADD_ELEMENT`;

export const ROTATE_ELEMENT = `${REDUCER_KEY}/ROTATE_ELEMENT`;
export const SELECT_ELEMENT = `${REDUCER_KEY}/SELECT_ELEMENT`;
export const SELECT_ELEMENTS = `${REDUCER_KEY}/SELECT_ELEMENTS`;
export const SET_AREA_COORDINATES = `${REDUCER_KEY}/SET_AREA_COORDINATES`;
export const SET_ELEMENT_COORDINATES = `${REDUCER_KEY}/SET_ELEMENT_COORDINATES`;
export const SET_ELEMENT_SIZES = `${REDUCER_KEY}/SET_ELEMENT_SIZES`;
export const SET_ELEMENTS_COORDINATES = `${REDUCER_KEY}/SET_ELEMENTS_COORDINATES`;
export const UPDATE_EVENTS_STATUS = `${REDUCER_KEY}/UPDATE_EVENTS_STATUS`;
export const UNSELECT_ELEMENT = `${REDUCER_KEY}/UNSELECT_ELEMENT`;
