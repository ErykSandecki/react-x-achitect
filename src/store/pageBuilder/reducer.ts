// others
import {
  ADD_ELEMENT,
  ADD_SELECTED_ELEMENT,
  REMOVE_SELECTED_ELEMENT,
  SET_SELECTED_ELEMENTS,
} from './actionsType';

// types
import { TAction } from 'types';
import {
  TAddELementAction,
  TAddSelectedElementAction,
  TPageBuilderState,
  TRemoveSelectedElementAction,
  TSetSelectedElementsAction,
} from './types';

// utils
import { handleAddElement } from './utils/handleAddElement';
import { omit } from 'lodash';

const initialState: TPageBuilderState = {
  elements: { dynamicData: {}, staticData: {} },
  isLoading: true,
  isPending: false,
  selectedElements: {},
};

const addElement = (
  state: TPageBuilderState,
  { payload: element }: TAction<TAddELementAction['payload']>,
): TPageBuilderState => handleAddElement(element, state);

const addSelectedElement = (
  state: TPageBuilderState,
  { payload: selectedElement }: TAction<TAddSelectedElementAction['payload']>,
): TPageBuilderState => ({
  ...state,
  selectedElements: {
    ...state.selectedElements,
    [selectedElement.id]: selectedElement,
  },
});

const removeSelectedElement = (
  state: TPageBuilderState,
  { payload: id }: TAction<TRemoveSelectedElementAction['payload']>,
): TPageBuilderState => ({
  ...state,
  selectedElements: omit(state.selectedElements, id),
});

const setSelectedElements = (
  state: TPageBuilderState,
  { payload: selectedElements }: TAction<TSetSelectedElementsAction['payload']>,
): TPageBuilderState => ({ ...state, selectedElements });

const pageBuilder = (
  state: TPageBuilderState = initialState,
  action: TAction,
): TPageBuilderState => {
  switch (action.type) {
    case ADD_ELEMENT:
      return addElement(state, action);
    case ADD_SELECTED_ELEMENT:
      return addSelectedElement(state, action);
    case REMOVE_SELECTED_ELEMENT:
      return removeSelectedElement(state, action);
    case SET_SELECTED_ELEMENTS:
      return setSelectedElements(state, action);
    default:
      return state;
  }
};

export default pageBuilder;
