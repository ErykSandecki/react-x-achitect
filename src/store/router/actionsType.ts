export const REDUCER_KEY = 'router';

export const HISTORY_CHANGED = `${REDUCER_KEY}/HISTORY_CHANGED`;
export const PUSH = `${REDUCER_KEY}/PUSH`;
export const REDIRECT_TO = `${REDUCER_KEY}/REDIRECT_TO`;
