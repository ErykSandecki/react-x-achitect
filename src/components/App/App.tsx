import { BrowserRouter as Router } from 'react-router-dom';
import { FC } from 'react';
import { useSelector } from 'react-redux';

// components
import AppHelmet from 'components/AppHelmet/AppHelmet';
import AppInitializer from 'components/AppInitializer/AppInitializer';
// import ErrorPage from 'pages/ErrorPage/ErrorPage';
// import { Notification, Snackbar } from 'components';

// hooks
import { useTheme } from 'hooks/useTheme/useTheme';

// core
import Routing from 'core/Routing/Routing';

// others
import { className, classNames } from './classNames';

// store
import {
  isAppLoadedSelector,
  isPendingSelector,
} from 'store/appInitializer/selectors';

// styles
import styles from './app.scss';

const App: FC = () => {
  const isAppLoaded = useSelector(isAppLoadedSelector);
  const isPending = useSelector(isPendingSelector);

  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  // if (!isAppLoaded && isPending) {
  //   return (
  //     <main className={cx(classNamesWithTheme[className])}>
  //       <AppInitializer />
  //     </main>
  //   );
  // }

  // if (!isAppLoaded && !isPending) {
  //   return <Router>{/* <ErrorPage /> */}</Router>;
  // }

  return (
    <main className={cx(classNamesWithTheme[className])}>
      <AppHelmet />
      <Router>
        <Routing />
      </Router>
      {/* <Notification />
      <Snackbar /> */}
    </main>
  );
};

export default App;
