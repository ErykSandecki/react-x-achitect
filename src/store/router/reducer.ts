// others
import { HISTORY_CHANGED, PUSH, REDIRECT_TO } from './actionsType';

// types
import { TAction } from 'types/redux';
import {
  THistoryChangedAction,
  TPushAction,
  TRedirectToAction,
  TRouterState,
} from './types';
import { RouterAction } from '../../types/router/enums';

const initialState: TRouterState = {
  action: RouterAction.push,
  location: undefined,
  redirectTo: undefined,
};

const historyChanged = (
  state: TRouterState,
  { payload: { action, location } }: TAction<THistoryChangedAction['payload']>,
): TRouterState => ({
  ...state,
  action,
  location,
  redirectTo: undefined,
});

const push = (
  state: TRouterState,
  { payload: to }: TAction<TPushAction['payload']>,
): TRouterState => ({
  ...state,
  redirectTo: {
    action: RouterAction.push,
    to,
  },
});

const redirectTo = (
  state: TRouterState,
  { payload: redirectTo }: TAction<TRedirectToAction['payload']>,
): TRouterState => ({
  ...state,
  redirectTo,
});

const routerReducer = (
  state: TRouterState = initialState,
  action: TAction,
): TRouterState => {
  switch (action.type) {
    case HISTORY_CHANGED:
      return historyChanged(state, action);
    case PUSH:
      return push(state, action);
    case REDIRECT_TO:
      return redirectTo(state, action);
    default:
      return state;
  }
};

export default routerReducer;
