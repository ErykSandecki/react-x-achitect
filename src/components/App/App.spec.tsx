import { HelmetProvider } from 'react-helmet-async';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import App from './App';

// mocks
import { wholeStateMock } from 'test/mocks/reducer/wholeStateMock';
import 'test/mocks/sagas/allSagas';

// others
import { REDUCER_KEY as APP_INITIALIZER } from 'store/appInitializer/actionsType';

// store
import { configureStore } from 'store/store';

jest.mock('core/Routing/Routing', () => () => 'Routing');

describe('App snapshots', () => {
  it('should render initializer', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render error page', () => {
    // mock
    const store = configureStore({
      ...wholeStateMock,
      [APP_INITIALIZER]: {
        isAppLoaded: false,
        isPending: false,
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render routing', () => {
    // mock
    const store = configureStore({
      ...wholeStateMock,
      [APP_INITIALIZER]: {
        isAppLoaded: true,
        isPending: false,
      },
    });

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <HelmetProvider>
          <App />
        </HelmetProvider>
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
