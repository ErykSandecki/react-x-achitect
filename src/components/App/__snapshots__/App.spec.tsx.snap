// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots should render error page 1`] = `
<DocumentFragment>
  <div
    class="Page Page--light"
  >
    <div
      class="ErrorPage ErrorPage--light"
    >
      <div
        class="ErrorPage__background ErrorPage__background--light"
      />
      <div
        class="ErrorPage__content ErrorPage__content--light"
      >
        <div
          class="ErrorPage__logo ErrorPage__logo--light"
          data-test-logo=""
        />
        <h3
          class="ErrorPage__title ErrorPage__title--light typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--regular__G4QAa typography_Typography--normal__2w6Rv typography_Typography--h3__yKIyB typography_Typography--without-margin__uJld2"
          data-test-text=""
          style="text-align: inherit;"
        >
          Interval Server Error
        </h3>
        <p
          class="ErrorPage__description ErrorPage__description--light typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--regular__G4QAa typography_Typography--normal__2w6Rv typography_Typography--p__efCcF typography_Typography--without-margin__uJld2"
          data-test-text=""
          style="text-align: inherit;"
        >
          The server has been deserted for a while. Please be patient or try again later.
        </p>
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App snapshots should render initializer 1`] = `
<DocumentFragment>
  <main
    class="App App--light"
  >
    <div
      class="AppInitializer"
    >
      <div
        class="AppInitializer__logo-wrapper"
      >
        <div
          class="AppInitializer__squares-wrapper"
        >
          <div
            class="AppInitializer__first-square-wrapper"
          >
            <div
              class="AppInitializer__first-square"
            />
          </div>
          <div
            class="AppInitializer__second-square-wrapper"
          >
            <div
              class="AppInitializer__second-square"
            />
          </div>
        </div>
        <div
          class="AppInitializer__logo"
        />
      </div>
      <h1
        class="AppInitializer__company-name typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--semi-bold__o4iwT typography_Typography--normal__2w6Rv typography_Typography--h1__NYzWT typography_Typography--without-margin__uJld2"
        data-test-text=""
        style="text-align: inherit;"
      >
        CryptoTradingHub
      </h1>
    </div>
  </main>
</DocumentFragment>
`;

exports[`App snapshots should render routing 1`] = `
<DocumentFragment>
  <main
    class="App App--light"
  >
    Routing
    <div
      class="notification_Notification__oqZza"
    />
    <div
      class="snackbar_Snackbar__fgwhT"
      data-test-snackbar=""
    >
      <svg
        class="name light"
        data-test-icon="icon-error-outlined"
        fill="none"
        height="16"
        style="flex-shrink: 0;"
        tabindex="-1"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          class=""
          d="M7.2 10.4H8.8V12H7.2V10.4ZM7.2 4H8.8V8.8H7.2V4ZM7.992 0C3.576 0 0 3.584 0 8C0 12.416 3.576 16 7.992 16C12.416 16 16 12.416 16 8C16 3.584 12.416 0 7.992 0ZM8 14.4C4.464 14.4 1.6 11.536 1.6 8C1.6 4.464 4.464 1.6 8 1.6C11.536 1.6 14.4 4.464 14.4 8C14.4 11.536 11.536 14.4 8 14.4Z"
          fill="#ef5350"
        />
      </svg>
      <div
        class="snackbar_Snackbar__content__XjQkT"
      >
        <small
          class="snackbar_Snackbar__message__hDlEl typography_Typography__Hvofe typography_Typography--light__ma6Vx typography_Typography--regular__G4QAa typography_Typography--normal__2w6Rv typography_Typography--small__S8-Gb typography_Typography--without-margin__uJld2"
          data-test-text=""
          style="text-align: inherit;"
        >
          snackbar.
        </small>
      </div>
      <svg
        class="snackbar_Snackbar__close-icon__yIufw index_IconClose__ccPvN index_IconClose--light__hk-9p index_IconClose--forced-hover--light__s2lnK"
        data-test-snackbar-close-icon="icon-close"
        fill="none"
        height="16"
        style="flex-shrink: 0;"
        tabindex="-1"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          class=""
          d="M8.00146 6.97108L14.9711 0L16 1.03038L9.02892 8.00146L16 14.9711L14.9711 16L8 9.02892L1.03038 16L0.00145541 14.9711L6.97253 8L0 1.03038L1.03038 0.00145532L8.00146 6.97253V6.97108Z"
          fill=""
        />
      </svg>
    </div>
  </main>
</DocumentFragment>
`;
